plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/BucketNameValidator.java

```diff
@@ -0,0 +1,52 @@
+package com.google.cloud.tools.eclipse.ui.util.databinding;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+
+import com.google.cloud.tools.eclipse.ui.util.Messages;
+
+/**
+ * Implements a simplified bucket name validation.
+ * <p>
+ * The following rules are verified:
+ * <p>
+ * Use lowercase letters, numbers, hyphens (-), and underscores (_). You can also use a dot (.) to form a valid
+ * top-level domain (e.g., example.com). Format: You must start and end the name with a number or letter. Bucket 
+ * names must contain 3 to 63 characters. Names containing dots can contain up to 222 characters, but each 
+ * dot-separated component can be no longer than 63 characters.
+ * <p>
+ * The actual rules that govern the bucket naming are more complex. See the complete list of bucket name requirements
+ * for more information: https://cloud.google.com/storage/docs/naming
```

Comment:
This should say whether we are more restrictive or more permissive than the naming standard.  I wonder if there's an approved regex already for this.